lib/collection/src/shards/transfer/shard_transfer.rs

```diff
@@ -316,6 +320,8 @@ async fn transfer_snapshot(
             shard_id,
             &shard_download_url,
             SnapshotPriority::ShardTransfer,
+            Some(SNAPSHOT_TRANSFER_RECOVER_TIMEOUT),
+            api::grpc::transport_channel_pool::DEFAULT_RETRIES,
```

Comment:
The timeout for the transfer/recovery process is now explicitly configured here as part of the logic driving the shard snapshot transfer.

Reply:
Why do we need to set it explicitly? Will it be configurable in the future? else I think we could avoid parameter drilling if they just depend on constants anyways

Reply:
Good point. I may have pulled them up a bit too far here.

I'll downgrade the timeout into `recover_shard_snapshot_from_url`, as I don't think it makes sense to move it further down into `with_shard_snapshot_client`. I've also removed the retries parameter.

See 44cbd9142c4ff567ff44375bb36bbf04f8a81223.